src/scheduler/server_info.cpp

```diff
@@ -798,6 +803,16 @@ query_server_dyn_res(server_info *sinfo)
 				log_eventf(PBSEVENT_DEBUG2, PBS_EVENTCLASS_SERVER, LOG_DEBUG, "server_dyn_res",
 					"%s = %s (\"%s\")", conf.dynamic_res[i].command_line, res_to_str(res, RF_AVAIL), buf);
 
+			if (pid != 0) {
```

Comment:
I think you want pid > 0.  If pid is negative, the fork() failed.

Reply:
Correct. If the fork fails, though, then we set pipe_err, but in this one we no longer have that guard.